e2e/tests/aou-elements-verify.spec.ts

```diff
@@ -0,0 +1,96 @@
+import {Browser, Page} from 'puppeteer';
+import NewClrIconLink from '../app/aou-elements/ClrIconLink';
+import GoogleLoginPage from '../app/GoogleLoginPage';
+import HomePage, {FIELD_LABEL} from '../app/HomePage';
+import PageNavigation, {LINK} from '../app/page-mixin/PageNavigation';
+import ProfilePage from '../app/ProfilePage';
+import WorkspaceCard from '../app/WorkspaceCard';
+import WorkspacesPage from '../app/WorkspacesPage';
+import launchBrowser from '../driver/puppeteer-launch';
+
+const configs = require('../resources/workbench-config');
+
+jest.setTimeout(60 * 1000);
+
+describe.skip('aou-elements', () => {
+
+  let browser: Browser;
+  let page: Page;
+
+  beforeAll(async () => {
+    browser = await launchBrowser();
+  });
+
+  afterAll(async () => {
+    await browser.close();
+  });
+
+  beforeEach(async () => {
+    const incognitoContext = await browser.createIncognitoBrowserContext();
+    page = await incognitoContext.newPage();
+    await page.setUserAgent(configs.puppeteerUserAgent);
+    const loginPage = new GoogleLoginPage(page);
+    await loginPage.login();
+  });
+
+  afterEach(async () => {
+    await page.close();
+    await page.waitFor(1000);
+  });
+
+  test('Workspace card', async () => {
+    const home = new HomePage(page);
+    await home.waitForReady();
+
+    await WorkspaceCard.getAllCards(page);
+    await WorkspaceCard.getAnyCard(page);
+
+    await PageNavigation.goTo(page, LINK.YOUR_WORKSPACES);
+    await new WorkspacesPage(page).waitForReady();
+
+    const n = 'aoutest-70-1583167646';
+    const myCard = await WorkspaceCard.findCard(page, n);
+    const myCardName = await myCard.getResourceCardName();
+    expect(myCardName).toEqual(n);
+    expect(await myCard.getEllipsisIcon()).toBeTruthy();
+
+    const linTexts = await myCard.getPopupLinkTextsArray();
+    expect(linTexts).toEqual(expect.arrayContaining(['Share', 'Edit', 'Duplicate', 'Delete']));
+  });
+
+  /**
+   * This is not a Puppeteer test for AoU application. It is for framework functions testing.
+   * If you make any change in aou-elements classes, you want to run this test to verify changes.
+   */
+  test('Click on Create New Workspace link on Home page', async () => {
```

Comment:
This looks like multiple test cases to me. Depending on how expensive a single test case setup is, I would split this out into at least a few targets categories of tests.

Reply:
I didn't pay too much attention to this test's organization and structure because it's used to test new or/and modified functions. I think there should be functions unit test cases to ensure functions don't break. I just haven't put in time to think it through more thoroughly. I need time to make a plan.

Reply:
removed `aou-elements-verify.spec.ts`